platform/default/src/mbgl/storage/offline_database.cpp

```diff
@@ -574,135 +718,172 @@ std::optional<std::pair<Response, uint64_t>> OfflineDatabase::getTile(const Reso
         size = data->length();
     }
 
-    return std::make_pair(response, size);
+    return std::make_pair(response, size); // Return the response and data size
 }
 
 std::optional<int64_t> OfflineDatabase::hasTile(const Resource::TileData& tile) {
-    // clang-format off
-    mapbox::sqlite::Query size{ getStatement(
-        "SELECT length(data) "
-        "FROM tiles "
-        "WHERE url_template = ?1 "
-        "  AND pixel_ratio  = ?2 "
-        "  AND x            = ?3 "
-        "  AND y            = ?4 "
-        "  AND z            = ?5 ") };
-    // clang-format on
+    // First, try to find the tile in the 'tiles' table
+    std::optional<int64_t> selectTilesResult = extractTileDataSize(
+        tile,
+        "SELECT length(data) FROM tiles WHERE url_template = ?1 AND pixel_ratio = ?2 AND x = ?3 AND y = ?4 AND z = ?5");
```

Comment:
This will always fail for users not using any offline functionality at all. I'm not comfortable proceeding without first ensuring the performance impact is minimal, otherwise we need to keep track of if the `tiles` table contains any tiles at all and skip this check.

Reply:
We agree with you on this point, a boolean variable which would be initialized on startup (by counting tiles / resources, or even by checking if there is an offline_region in the database or not) and set when we add the first resource / tile is a good idea in our opinion, we will try to implement that.